flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -339,6 +351,11 @@ Builder planParallelism(Integer parallelism) {
       return this;
     }
 
+    Builder monitorSnapshotNumber(int newMonitorSnapshotNumber) {
+      this.monitorSnapshotNumber = newMonitorSnapshotNumber;
```

Comment:
Should we add a check here? It should be greater than 0, right?

Reply:
+1. A precondition check. And INT_MAX can be the value that disables this behavior.

Or a negative value (eg -1). A negative value is probably more in-line with what we typically do and more cross-language friendly (as Iceberg table format is a specification first and foremost… it should be able to be rewritten in a language that doesn’t have JVM INT_MAX).